protocol_decoder/src/decoding.rs

```diff
@@ -350,12 +354,26 @@ impl ProcessedBlockTrace {
                 let txn_idx_of_dummy_entry =
                     txn_ir.last().unwrap().txn_number_before.low_u64() as usize + 1;
```

Comment:
Why are we guaranteed that `txn_ir` is never empty? And why wouldn't `txn_ir.last()?.txn_number_before.low_u64()` work?